raft/server.cc

```diff
@@ -1888,4 +1892,27 @@ std::ostream& operator<<(std::ostream& os, const server_impl& s) {
     return os;
 }
 
+future<> server_impl::override_snapshot_thresholds() {
+    return utils::get_local_injector().inject(
+        "raft_server_set_snapshot_thresholds", [this](auto& handler) -> future<> {
+            const auto set_parm = [&handler](auto& target, const std::string_view name) -> void {
+                const auto from = handler.get(name);
+                if (from) {
+                    try {
+                        target = boost::lexical_cast<std::remove_reference_t<decltype(target)>>(*from);
+                        logger.info("Applied _config.{}={}", name, *from);
+                    } catch (const boost::bad_lexical_cast& e) {
+                        logger.info("Could not apply a snapshot threshold param: {}, value: {}, error: {}",
+                                    name, *from, e.what());
+                    }
```

Comment:
I think we should rethrow the exception here. If we get this exception, then there is something wrong with a test that uses this error injection (not necessarily the test you added; different tests could you this injection in the future). We want to ensure that in this case the test fails and it's easy to see why.